docs/.markdownlint.yml

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# markdownlint-cli configuration file.
+# https://github.com/igorshubovych/markdownlint-cli
+# https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md
+
+# Enable all markdownlint rules by default
+default: true
+
+# List formatting:
+#
+# 1. Lists must have a blank line before AND after the list.
+# 2. Lists start aligned to the left (do not indent the top level list items).
+#    NOTE: markdownlint currently checks indentation for unordered lists only.
+#          Please manually verify that your ordered lists are not indented.
+#          See https://github.com/DavidAnson/markdownlint/issues/138.
+# 3. You may use one or zero blank lines between list items.
+# 4. Nested list items should be indented to align with the first character of
+#    the first line. For bullet lists, that means 2 spaces. For numbered
+#    lists, that's 3 spaces (but 4 spaces is okay if that's easier).
+# 5. In multiline list items, subsequent lines are indented by 2 spaces.
```

Comment:
I've not seen that before. I don't really like it and it actually doesn't match the changes you've made here. I prefer this, which I think matches Google's markdown formatter

```markdown
* Some list item that is really long or
  whatever.
* Another list item
```

as opposed to what I think this is suggesting, which would be

```markdown
* Some list item that is really long or
    whatever.
* Another list item
```

That just looks weird to me.

Oh but that's actually not what the examples show. Ok I don't think this is worded clearly.

```suggestion
# 5. In multiline list items, subsequent lines are also indented to align with
#   the first character of the first line.
```

Reply:
I mostly copied this comment from https://github.com/openxla/stablehlo/blob/14ba2bae5549d662fd761c031d6db08930b21392/.markdownlint.yaml#L21-L47 , but then I made some changes and didn't update the comment.

The [`ul-indent` rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md007---unordered-list-indentation) only specifies how much to indent between nested levels of lists:
```markdown
<!-- It has opinions on this -->
* List item
  * Nested list item indented by 2 spaces

<!-- It does not have opinions on this -->
* List item that
is split over multiple lines
* Another list item that
  is split over multiple lines
```

We don't have many nested lists (for good reason IMO), but the few we had were using 4 spaces, so I kept that.

As for indentation at the same nest level, agreed about aligning, and I manually updated the formatting in affected files to match that.